lib/makefile

```
@@ -0,0 +1,39 @@
+# Copyright 2020-2022 CYBERCRYPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+##### Help message #####
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make <target> \033[36m\033[0m\n\nTargets:\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2 }' $(MAKEFILE_LIST)
+
+##### Build targets #####
+.PHONY: build
+build: ## Build the Encryptonize library
+	go get -v ./...
+	go build -v .
+
+.PHONY: lint
+lint: ## Lint the codebase
+	echo "[*] formatting code"
+	gofmt -l -w .
+
+	echo "[*] tidying up"
+	go mod tidy
+
+	echo "[*] running linter"
+	golangci-lint run
+
+##### Test targets #####
+.PHONY: tests
+tests: ## Run tests
+	go test -count=1
```

This doesn't run tests on the nested packages.
```suggestion
	go test -count=1 ./...
```